CompilerSource/compiler/components/write_defragged_events.cpp

```diff
@@ -114,17 +154,18 @@ int lang_CPP::compile_writeDefraggedEvents(const GameData &game, const ParsedObj
 
   wto << "  struct event_parent: " << system_get_uppermost_tier() << endl;
   wto << "  {" << endl;
-            for (evfit it = used_events.begin(); it != used_events.end(); it++)
-            {
-              const bool e_is_inst = event_is_instance(it->second.mid, it->second.id);
-              if (event_has_sub_check(it->second.mid, it->second.id) && !e_is_inst) {
-                wto << "    inline virtual bool myevent_" << it->first << "_subcheck() { return false; }\n";
-              }
-              wto << (e_is_inst ? "    virtual void    myevent_" : "    virtual variant myevent_") << it->first << "()";
-              if (event_has_default_code(it->second.mid,it->second.id))
-                wto << endl << "    {" << endl << "  " << event_get_default_code(it->second.mid,it->second.id) << endl << (e_is_inst ? "    }" : "    return 0;\n    }") << endl;
-              else wto << (e_is_inst ? " { } // No default " : " { return 0; } // No default ") << event_get_human_name(it->second.mid,it->second.id) << " code." << endl;
-            }
+  for (const auto &str_ev : used_events.base_methods) {
+    const string &fname = str_ev.first;
+    const EventDescriptor &event = str_ev.second;
+    const bool e_is_inst = event.IsInstance();
+    if (event.HasSubCheck() && !e_is_inst) {
+      wto << "    inline virtual bool myevent_" << fname << "_subcheck() { return false; }\n";
```

Comment:
I know this probably the best and maybe the only way to generate our code but it's so damn ugly. I wish we had some sort of template files we could subst instead so it was clearer what the end file should look like

Reply:
That's totally doable with string interpolation. I mean even LGM does that with messages.properties.
https://stackoverflow.com/questions/37956090/string-interpolation-in-c-construct-a-stdstring-with-embedded-values-e-g